Wikipedia/UI-V5/WMFTableOfContentsAnimator.swift

```diff
@@ -0,0 +1,59 @@
+
+import UIKit
+
+class WMFTableOfContentsAnimator: NSObject, UIViewControllerAnimatedTransitioning {
+    
+    // MARK: - init
+    let isPresenting :Bool
```

Comment:
@bgerstle Old discussion… same rationale still stands: 
- It is pretty much convention for the entire internet to do these in the same class. 
- …which It makes sense… if not for code sharing (which can happen frequently), then for logic grouping (would you implement viewDidAppear and viewDidDisappear in 2 different classes?)

Just because this keeps coming up, I will dive a bit further so we can get it settled and avoid repeat discussions:
**Current Implementation**: the view controller figures out if it is presenting or not and sets a Bool on a class
**Your proposal**: Add a class. the view controller figures out if it is presenting or not, and vends a specific class.

In your proposal, the point of the decision hasn't changed at all: the responsibility of figuring out whether we are presenting or dismissing is still on the same view controller.

Additionally, no code was saved. The code isn't safer. It isn't more obvious. It doesn't have better performance. 

On the other side, it doubles the amount of animation classes, and goes against the convention the rest of the Cocoa world has settled on.

So for me suggesting this class be separated to prevent setting a single Bool on init seems like mostly bike shedding on something that doesn't actually benefit the code base.

If you want to discus further let me know… but to me it is pretty clear why we shouldn't go the way you are suggestion. Either way lets close the book on it.